lib/refile/custom_logger.rb

```diff
@@ -31,5 +29,14 @@ def log(env, status, began_at)
         (now - began_at) * 1000
       )
     end
+
+    def logger
+      @logger ||= @logger_proc.call
```

Comment:
Was a bit fast on reviewing this. Lazily initializing like this is not thread safe. Probably not a huge deal, but it could lead to garbled logs in some rare cases.